cli/src/main/java/org/ehrbase/cli/config/CliAqlQueryContext.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2019-2024 vitasystems GmbH.
+ *
+ * This file is part of project EHRbase
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehrbase.cli.config;
+
+import java.net.URI;
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import org.ehrbase.api.dto.AqlQueryContext;
+import org.ehrbase.api.service.StatusService;
+import org.ehrbase.openehr.sdk.response.dto.MetaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * AQL query context implementation used for EHRbase CLI that can be configured using
+ * <code>-aql-mode=default|dry_run|show_executed_aql|show_executed_sql|show_query_plane</code> params.
+ */
+@Component(AqlQueryContext.BEAN_NAME)
+public class CliAqlQueryContext implements AqlQueryContext {
```

Comment:
This should just throw `UnsupportedOperationException` on any method call, since executing AQL is not a feature of the CLI

Reply:
Was intended as a preparation in case we need cli AQL support. I have replaced the method bodies with `UnsupportedOperationException`.